pkg/etcdutil/etcdutil.go

```diff
@@ -294,3 +321,87 @@ func TakeAndSaveFullSnapshot(ctx context.Context, client client.MaintenanceClose
 
 	return snapshot, nil
 }
+
+// checkFullSnapshotIntegrity verifies the integrity of the full snapshot by comparing
+// the appended SHA256 hash of the full snapshot with the calculated SHA256 hash of the full snapshot data.
+func checkFullSnapshotIntegrity(snapshotData io.ReadCloser, snapTempDBFilePath string, logger *logrus.Entry) (io.ReadCloser, error) {
+	logger.Info("checking the full snapshot integrity with the help of SHA256")
+
+	// If previous temp db file already exist then remove it.
+	if err := os.Remove(snapTempDBFilePath); err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+
+	db, err := os.OpenFile(snapTempDBFilePath, os.O_RDWR|os.O_CREATE, 0600)
```

Comment:
Missing `db.Close()` after opening the file. I see that the `db` file is returned as is at the end of the function, so can you please add a note in the docstring for this function for the caller to take care of closing the file once they read it? I don't see a `snapshotData.Close()` in the caller function as well, but just a `os.Remove()` of the file, which is not sufficient, since that can lead to memory leaks.